src/Mod/Path/PathScripts/PathProfile.py

```diff
@@ -322,12 +324,16 @@ def areaOpShapes(self, obj):
 
         # Pre-process Base Geometry to process edges
         if obj.Base and len(obj.Base) > 0:  # The user has selected subobjects from the base.  Process each.
-            shapes.extend(self._processEdges(obj))
+            edge_shapes, otherBase = self._processEdges(obj)
+            shapes.extend(edge_shapes)
 
-        if obj.Base and len(obj.Base) > 0:  # The user has selected subobjects from the base.  Process each.
+        if otherBase is False:
+            # Edges were already processed, or whole model targeted.
+            PathLog.debug("otherBase is False")
+        elif otherBase and len(otherBase) > 0:  # The user has selected subobjects from the base.  Process each.
             if obj.EnableRotation != 'Off':
-                for p in range(0, len(obj.Base)):
-                    (base, subsList) = obj.Base[p]
+                for p in range(0, len(otherBase)):
+                    (base, subsList) = otherBase[p]
```

Comment:
I gotta say the old code was more readable, it's pretty clear what `obj.Base` is, it's not clear what `otherBase` is though - makes it hard to understand what's going on.

Reply:
Normally, yes, `obj.Base` would be used, as it was before these changes.  However there are related issues that these changes fix.
.
First, I learned that `obj` attributes should, for the most part, be read-only - so as to not introduce errors into the operation.  
.
Second, because the Profile op is handling edges and faces, it is processing those two feature sets(selections) separately.  Above, near line 325 the `obj.Base` object is pre-processed to extract and process edges.  
I am using the new `otherBase` as a container identical to `obj.Base` to contain features that were not handled in the edge processing section and need to be processed in the next face-processing section.
In the edges processing code, edges are removed from the selection(normally `obj.Base`) and processed into removal shapes, which are returned as part of the tuple from ` self._processEdges(obj)` call.  The other object returned in the tuple is the remaining list of features provided in `obj.Base` that still need processing.  I am storing this remainder in the new `otherBase` object because I don't know of another way to easily flag or mark those edge features within `obj.Base` that are already processed and now need to be ignored.  Creating the `newBase` and corresponding `otherBase` is an easy way to return these unprocessed features without having to filter out already processed edge features.  And, I am not modifying `obj.Base` as stated in my first declaration.
.
Yes, now that you mention it, the `newBase` and `otherBase` should carry the same name so readers see that they are one in the same.  Yes, I am open for a better name, like `remainingBaseGeometry`.  I am open to ideas.
.
The overall fix does work.  From my limited perspective it works fairly well.  Yes, I need to improve readability, and will work to do so.  Ideas to improve?